pytorch_tcn/tcn.py

```diff
@@ -129,161 +131,60 @@ def get_kernel_init_fn(
 
 
 
-class CausalConv1d(nn.Conv1d):
+class BaseTCN(nn.Module):
     def __init__(
             self,
-            in_channels,
-            out_channels,
-            kernel_size,
-            stride = 1,
-            dilation = 1,
-            groups = 1,
-            bias = True,
-            buffer = None,
-            lookahead = 0,
-            **kwargs,
             ):
-        
-        super(CausalConv1d, self).__init__(
-            in_channels = in_channels,
-            out_channels = out_channels,
-            kernel_size = kernel_size,
-            stride = stride,
-            padding = 0,
-            dilation = dilation,
-            groups = groups,
-            bias = bias,
-            **kwargs,
-            )
-        
-        self.pad_len = (kernel_size - 1) * dilation
-        if lookahead > self.pad_len//2:
-            lookahead = self.pad_len//2
-        self.lookahead = lookahead
-
-        self.buffer_len = self.pad_len - self.lookahead
-        #print( 'pad len:', self.pad_len )
-        #print( 'lookahead:', self.lookahead )
-        #print( 'buffer len:', self.buffer_len )
-        
-        if buffer is None:
-            buffer = torch.zeros(
-                1,
-                in_channels,
-                self.pad_len,
-                )
-            
-        self.register_buffer(
-            'buffer',
-            buffer,
-            )
-        
+        super(BaseTCN, self).__init__()
         return
     
-    def _forward(self, x):
-        p = nn.ConstantPad1d(
-            ( self.buffer_len, self.lookahead ),
-            0.0,
-            )
-        x = p(x)
-        x = super().forward(x)
-        return x
-
-    def forward(
+    def inference(
             self,
             x,
-            inference=False,
+            in_buffers=None,
```

Comment:
should be:

def inference( *args, **kwargs ):
x = self( args, kwargs, inference=True )
return x

Reply:
Done.